app/lib/actions/pulp/repository/destroy.rb

```diff
@@ -8,10 +8,13 @@ class Destroy < Pulp::AbstractAsyncTask
         end
 
         def invoke_external_task
-          output[:pulp_tasks] = pulp_extensions.repository.delete(input[:pulp_id])
-        rescue RestClient::ResourceNotFound
-          Rails.logger.warn("Tried to delete repository #{input[:pulp_id]}, but it did not exist.")
-          []
+          repo = ::Katello::Repository.find_by(:pulp_id => input[:pulp_id])
```

Comment:
we want to change this to just use 'repository_id', not pulp_id.  This will allow it to potentially work with things like pulp3, although likely more changes are needed.

Reply:
getting an error when deleting a content view version, because the ContentViewPuppetEnvironment isn't actually a Repository object.  It may be easier to explain in person, and there are a couple options, but something like:

```
if input[:content_view_puppet_environment_id]
  repo = ::Katello::ContentViewPuppetEnvironment.find(input[:content_view_puppet_environment_id]).nonpersisted_repository
else
  repo = ::Katello::Repository.find(input[:repository_id])
end
```